pkg/grpc/universalapi/api_metadata.go

```diff
@@ -143,9 +142,12 @@ func metadataConvertPubSubSubscriptionRules(rules []*runtimePubsub.Rule) *runtim
 		Rules: make([]*runtimev1pb.PubsubSubscriptionRule, 0),
 	}
 	for _, r := range rules {
+		match := ""
```

Comment:
Since you're at it, please change this method to pre-allocate the slice and improve the memory allocation:

```go
func metadataConvertPubSubSubscriptionRules(rules []*runtimePubsub.Rule) *runtimev1pb.PubsubSubscriptionRules {
	out := &runtimev1pb.PubsubSubscriptionRules{
		Rules: make([]*runtimev1pb.PubsubSubscriptionRule, len(rules)),
	}
	for i, r := range rules {
		out.Rules[i] = &runtimev1pb.PubsubSubscriptionRule{
			Path:  r.Path,
		}
		if r.Match != nil {
			out.Rules[i].Match = r.Match.String()
		}
	}
	return out
}
```

Reply:
Done.